client-v2/src/main/java/com/clickhouse/client/api/insert/InsertResponse.java

```diff
@@ -0,0 +1,25 @@
+package com.clickhouse.client.api.insert;
+
+import com.clickhouse.client.ClickHouseClient;
+import com.clickhouse.client.ClickHouseResponse;
+import com.clickhouse.client.ClickHouseResponseSummary;
+
+public class InsertResponse implements AutoCloseable {
+    private final ClickHouseResponse responseRef;
+    private final ClickHouseClient client;
+
+    public InsertResponse(ClickHouseClient client, ClickHouseResponse responseRef) {
+        this.responseRef = responseRef;
+        this.client = client;
+    }
+
+    public ClickHouseResponseSummary getSummary() {
+        return responseRef.getSummary();
+    }
+
+    @Override
+    public void close() {
+        responseRef.close();
```

Comment:
I would wrap with try-catch so responseRef.close() exception would not cause client unclosed